internal/controller/k8s_resources.go

```diff
@@ -99,10 +107,101 @@ func convertMetaObjectToMetadata(logger *zap.SugaredLogger, ctx context.Context,
 			return objMetadata, nil
 		}
 		objMetadata.KindSpecific = &pb.KubernetesObjectData_Node{Node: &pb.KubernetesNodeData{ProviderId: providerId}}
+	case "Service":
+		convertedServiceData, err := convertToKubernetesServiceData(ctx, obj.GetName(), clientset, obj.GetNamespace())
+		if err != nil {
+			return objMetadata, nil
+		}
+		objMetadata.KindSpecific = &pb.KubernetesObjectData_Service{Service: convertedServiceData}
+
 	}
 	return objMetadata, nil
 }
 
+func convertIngressToStringList(ingresses []v1.LoadBalancerIngress) []string {
+	result := []string{}
+	for _, ingress := range ingresses {
+		if ingress.IP != "" {
+			result = append(result, ingress.IP)
+		} else if ingress.Hostname != "" {
```

Comment:
This will fix your failing test case in `TestConvertIngressToStringList`:
```suggestion
		}
		if ingress.Hostname != "" {
```